core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1138,128 @@ def delete_playthroughs_multi(cls, playthrough_ids):
+    interaction_id = ndb.StringProperty(required=True, indexed=True)
+    # List of LearnerAnswerInfo dicts, which is defined in
+    # stats_domain.py, each dict corresponds to a single answer info of
+    # learner.
+    learner_answer_info_list = ndb.JsonProperty(repeated=True, indexed=False)
+    # The version of the learner_answer_info_list currently supported by Oppia.
+    # If the internal JSON structure of learner_answer_info_list changes,
+    # CURRENT_SCHEMA_VERSION in this class needs to be incremented.
+    schema_version = ndb.IntegerProperty(
+        indexed=True, default=(
+            feconf.CURRENT_LEARNER_ANSWERS_DETAILS_SCHEMA_VERSION))
+    # The total number of bytes needed to store all of the answers in the
+    # learner_answer_info_list. This value is found by summing the JSON
+    # sizes of all answer info dicts stored inside learner_answer_info_list.
+    accumulated_answer_info_json_size_bytes = ndb.IntegerProperty(
+        indexed=True, required=False, default=0)
+
+    @classmethod
+    def get_state_reference_for_exploration(cls, exp_id, state_name):
+        """Generate the state_reference for the state in an exploration.
+
+        Args:
+            exp_id: str. The id of the exploration.
+            state_name: str. The name of the state.
+
+        Returns:
+            str. The state_reference for a new instance of this class.
+        """
+        return '%s.%s' % (exp_id, state_name)
+
+    @classmethod
+    def get_state_reference_for_question(cls, question_id):
+        """Generate the state_reference for the state in the question.
+
+        Args:
+            question_id: str. The id of the question.
+
+        Returns:
+            str. The state_reference for a new instance of this class.
+        """
+        return question_id
+
+    @classmethod
+    def create_model_instance(
+            cls, state_reference, entity_type, interaction_id,
+            learner_answer_info_list, schema_version,
+            accumulated_answer_info_json_size_bytes):
+        """Creates a new LearnerAnswerDetailsModel for the given entity type
+        then writes it to the datastore.
+
+        Args:
+            state_reference: str. The reference to the state for which model
+                instance is being created. For exploration state it will be of
+                the form 'exp_id.state_name', & for question it will be of the
+                form 'question_id'.
+            entity_type: str. The type of entity i.e "exploration" or
+                "question".
+            interaction_id: str.  The ID of the interaction for which the
+                answer details are received.
+            learner_answer_info_list: list. The list of LearnerAnswerInfo
+                objects in dict format, which is defined in the stats_domain.
+            schema_version: int. The version of LearnerAnswerInfo dict, which
+                is currently supported by the Oppia.
```

Comment:
No, this should be the schema version of the LearnerAnswerInfo dict. It might be older than the "current" schema version if we change the structure in the future.

Reply:
Done